spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/DirectMessageListenerContainer.java

```diff
@@ -95,59 +100,155 @@ public void setConsumersPerQueue(int consumersPerQueue) {
 	}
 
 	/**
-	 * Set to false to requeue failed deliveries.
-	 * @param requeueRejected the requeue rejected.
+	 * Set to true for an exclusive consumer - if true, the
+	 * {@link #setConsumersPerQueue(int) consumers per queue} must be 1.
+	 * @param exclusive true for an exclusive consumer.
 	 */
-	public void setDefaultRequeueRejected(boolean requeueRejected) {
-		this.defaultRequeueRejected = requeueRejected;
+	@Override
+	public final void setExclusive(boolean exclusive) {
+		Assert.isTrue(!exclusive || (this.consumersPerQueue == 1),
+				"When the consumer is exclusive, the consumers per queue must be 1");
+		super.setExclusive(exclusive);
+	}
+
+	@Override
+	public void addQueueNames(String... queueNames) {
+		addQueues(Arrays.asList(queueNames));
+		super.addQueueNames(queueNames);
+	}
+
+	@Override
+	public void addQueues(Queue... queues) {
+		addQueues(Arrays.asList(queues)
+				.stream()
+				.map(Queue::getName)
+				.collect(Collectors.toList()));
+		super.addQueues(queues);
+	}
+
+	private void addQueues(Collection<String> queues) {
+		if (isRunning()) {
+			synchronized (this.consumersMonitor) {
+				Set<String> current = getQueueNamesAsSet();
+				for (String queue : queues) {
+					if (current.contains(queue)) {
+						throw new IllegalStateException("Queue " + queue + " is already configured for this container: "
+								+ this + ", no queues added");
+					}
+				}
+				List<String> added = new ArrayList<>();
+				for (String queue : queues) {
+					try {
+						consumeFromQueue(queue);
+						added.add(queue);
+					}
+					catch (IOException e) {
+						logger.error("Failed to add queue: " + queue + " undoing adds (if any)");
```

Comment:
Ha! Because `logger` is a property in super class.
I guess that broader area is just not supported.

Not sure how that would be easy to trace properties from super classes without compilation...

I guess we don't have choice unless live with this limitation.